mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1065,12 +1065,9 @@ def _generate_model_predictions(self):
             self.is_binomial = self.num_classes <= 2
 
             if self.is_binomial:
-                if list(self.label_list) not in [[0, 1], [-1, 1]]:
-                    raise ValueError(
-                        "Binary classifier evaluation dataset positive class label must be 1 or"
-                        " True, negative class label must be 0 or -1 or False, and dataset"
-                        " must contains both positive and negative examples."
-                    )
+                if self.pos_label in self.label_list:
+                    self.label_list.remove(self.pos_label)
```

Comment:
```suggestion
                    self.label_list = np.delete(self.label_list, np.where(self.label_list == self.pos_label))
```
I think numpy returns a copy of the immutable array and deletion is performed through index position location.

Reply:
Great catch! ndarray doesn't even define `remove()`. Not sure how this was succeeding. Will revert the other 2.0 PR and fix this :)

Reply:
I'm working on a branch for 2.0

Reply:
almost ready